app/src/main/java/com/danlls/daniel/todule_android/activities/ToduleListFragment.java

```diff
@@ -250,16 +252,42 @@ public void onSaveInstanceState(Bundle outState) {
                 break;
         }
         if (mCurFilter != null){
-            select += " AND (" + TodoEntry.COLUMN_NAME_TITLE + " LIKE ?)";
-            selectionArgs.add("%" + mCurFilter + "%");
+
+              Uri LABEL_URI = ToduleDBContract.TodoLabel.CONTENT_URI;
+              String labelSelect = "(" + ToduleDBContract.TodoLabel.COLUMN_NAME_TAG + " LIKE ?)";
+              Cursor labelCr = getContext().getContentResolver().query(LABEL_URI, ToduleDBContract.TodoLabel.PROJECTION_ALL, labelSelect,new String[]{"%" + mCurFilter + "%"},null);
+              ArrayList<String> labelIdList = new ArrayList<>();
+              while (labelCr.moveToNext()){
+                  labelIdList.add(labelCr.getString(labelCr.getColumnIndexOrThrow(ToduleDBContract.TodoLabel._ID)));
+              }
+
+              String idValue = listToString(labelIdList);
+              select += " AND ((" + TodoEntry.COLUMN_NAME_TITLE + " LIKE ?)";
+              selectionArgs.add("%" + mCurFilter + "%");
+              if (idValue != null)
+              select += " OR (" + TodoEntry.COLUMN_NAME_LABEL+ " IN ("+idValue+")))";
```

Comment:
Generally, when writing SQL queries involving inputs, direct string concatenation with user input should be avoided. Instead use parameterized query with placeholder to prevent SQL injection attacks.

How to fix:
- Replace `idValue` in `select` with placeholders `?`. On how to generate arbitrary number of placeholders, you can refer to https://stackoverflow.com/questions/7418849/in-clause-and-placeholders
- Add all the corresponding inputs into `selectionArgs`.